### ISSUE TITLE
[FIX] Unable to run the training job

### ISSUE THREAD

**Issue by tomcorten** [fmind/mlops-python-package#16]
## Bug Description

I have used your template before (approx 1 year ago) without issues. Now, I re-cloned your project, installed it's dependencies, and I am trying to run the training job: `poetry run bikes confs/training.yaml`. Whenever I try to do this, I get the following: 
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/tomcorten/Documents/personal_learning/mlops-python-package/src/bikes/scripts.py", line 46, in main
    runner.run()
  File "/Users/tomcorten/Documents/personal_learning/mlops-python-package/src/bikes/jobs/training.py", line 136, in run
    self.alerts_service.notify(
  File "/Users/tomcorten/Documents/personal_learning/mlops-python-package/src/bikes/io/services.py", line 116, in notify
    notification.notify(
  File "/Users/tomcorten/Documents/personal_learning/mlops-python-package/.venv/lib/python3.12/site-packages/plyer/facades/notification.py", line 84, in notify
    self._notify(
  File "/Users/tomcorten/Documents/personal_learning/mlops-python-package/.venv/lib/python3.12/site-packages/plyer/facades/notification.py", line 93, in _notify
    raise NotImplementedError("No usable implementation found!")
NotImplementedError: No usable implementation found!
```
## Expected Behavior

I am unfamiliar with the `plyer` package, but I assume that I can expect a message about the training job being finished.

## Steps to Reproduce

1. Installation
- clone
- install
2. Run: `poetry run bikes confs/training.yaml`

## Additional Context

**MacOS**
ProductName:            macOS
ProductVersion:         13.2.1
BuildVersion:           22D68

**Python**
Python 3.12.5

**Poetry**
```
absl-py                            2.1.0           Abseil Python Common Libraries, see https://github.com/abseil/abseil-py.
alembic                            1.13.2          A database migration tool for SQLAlchemy.
aniso8601                          9.0.1           A library for parsing ISO 8601 strings.
annotated-types                    0.7.0           Reusable constraint types to use with typing.Annotated
anyio                              4.4.0           High level compatibility layer for multiple asynchronous event loop imple...
azure-core                         1.31.0          Microsoft Azure Core Library for Python
azure-storage-blob                 12.22.0         Microsoft Azure Blob Storage Client Library for Python
azure-storage-file-datalake        12.16.0         Microsoft Azure File DataLake Storage Client Library for Python
black                              24.2.0          The uncompromising code formatter.
blinker                            1.8.2           Fast, simple object-to-object and broadcast signaling
boto3                              1.35.20         The AWS SDK for Python
botocore                           1.35.20         Low-level, data-driven core of boto 3.
bridgecraft                        0.13.0          
cachetools                         5.5.0           Extensible memoizing collections and decorators
category-encoders                  2.6.3           A collection of sklearn transformers to encode categorical variables as n...
certifi                            2024.8.30       Python package for providing Mozilla's CA Bundle.
cffi                               1.17.1          Foreign Function Interface for Python calling C code.
cfgv                               3.4.0           Validate configuration and produce human readable error messages.
charset-normalizer                 3.3.2           The Real First Universal Charset Detector. Open, modern and actively main...
clarabel                           0.9.0           Clarabel Conic Interior Point Solver for Rust / Python
click                              8.1.7           Composable command line interface toolkit
cloudpickle                        3.0.0           Pickler class to extend the standard pickle.Pickler functionality
contourpy                          1.3.0           Python library for calculating contours of 2D quadrilateral grids
coverage                           7.6.1           Code coverage measurement for Python
crossbricks                        0.12.0          
cryptography                       43.0.1          cryptography is a package which provides cryptographic recipes and primit...
cvxpy                              1.5.3           A domain-specific language for modeling convex optimization problems in P...
cycler                             0.12.1          Composable style cycles
databricks-connect                 15.1.0          Databricks Connect Client
databricks-sdk                     0.29.0          Databricks SDK for Python (Beta)
databricks-sql-connector           3.4.0           Databricks SQL Connector for Python
deprecated                         1.2.14          Python @deprecated decorator to deprecate old python classes, functions o...
distlib                            0.3.8           Distribution utilities
docker                             7.1.0           A Python library for the Docker Engine API.
ecos                               2.0.14          This is the Python package for ECOS: Embedded Cone Solver. See Github pag...
et-xmlfile                         1.1.0           An implementation of lxml.xmlfile for the standard library
filelock                           3.16.0          A platform independent file lock.
flask                              3.0.3           A simple framework for building complex web applications.
fonttools                          4.53.1          Tools to manipulate font files
gitdb                              4.0.11          Git Object Database
gitpython                          3.1.43          GitPython is a Python library used to interact with Git repositories
google-api-core                    2.19.2          Google API client core library
google-auth                        2.34.0          Google Authentication Library
google-cloud-core                  2.4.1           Google Cloud API client core library
google-cloud-storage               2.18.2          Google Cloud Storage API client library
google-crc32c                      1.6.0           A python wrapper of the C library 'Google CRC32C'
google-resumable-media             2.7.2           Utilities for Google Media Downloads and Resumable Uploads
googleapis-common-protos           1.65.0          Common protobufs used in Google APIs
graphene                           3.3             GraphQL Framework for Python
graphql-core                       3.2.4           GraphQL implementation for Python, a port of GraphQL.js, the JavaScript r...
graphql-relay                      3.2.0           Relay library for graphql-core
grpcio                             1.66.1          HTTP/2-based RPC framework
grpcio-status                      1.66.1          Status proto mapping for gRPC
gunicorn                           23.0.0          WSGI HTTP Server for UNIX
h11                                0.14.0          A pure-Python, bring-your-own-I/O implementation of HTTP/1.1
httpcore                           1.0.5           A minimal low-level HTTP client.
httpx                              0.27.2          The next generation HTTP client.
identify                           2.6.1           File identification library for Python
idna                               3.10            Internationalized Domain Names in Applications (IDNA)
imbalanced-learn                   0.12.3          Toolbox for imbalanced dataset in machine learning.
imblearn                           0.0             Toolbox for imbalanced dataset in machine learning.
immutabledict                      4.2.0           Immutable wrapper around dictionaries (a fork of frozendict)
importlib-metadata                 8.5.0           Read metadata from Python packages
inflect                            7.4.0           Correctly generate plurals, singular nouns, ordinals, indefinite articles
iniconfig                          2.0.0           brain-dead simple config-ini parsing
isodate                            0.6.1           An ISO 8601 date/time/duration parser and formatter
itsdangerous                       2.2.0           Safely pass data to untrusted environments and back.
jinja2                             3.1.4           A very fast and expressive template engine.
jmespath                           1.0.1           JSON Matching Expressions
joblib                             1.4.2           Lightweight pipelining with Python functions
kiwisolver                         1.4.7           A fast implementation of the Cassowary constraint solver
lz4                                4.3.3           LZ4 Bindings for Python
mako                               1.3.5           A super-fast templating language that borrows the best ideas from the exi...
markdown                           3.7             Python implementation of John Gruber's Markdown.
markupsafe                         2.1.5           Safely add untrusted strings to HTML/XML markup.
matplotlib                         3.9.2           Python plotting package
mlflow                             2.16.2          MLflow is an open source platform for the complete machine learning lifec...
mlflow-skinny                      2.16.2          MLflow is an open source platform for the complete machine learning lifec...
more-itertools                     10.5.0          More routines for operating on iterables, beyond itertools
mypy-extensions                    1.0.0           Type system extensions for programs checked with the mypy type checker.
nodeenv                            1.9.1           Node.js virtual environment builder
nptyping                           2.5.0           Type hints for NumPy.
numpy                              1.26.4          Fundamental package for array computing in Python
oauthlib                           3.2.2           A generic, spec-compliant, thorough implementation of the OAuth request-s...
openpyxl                           3.1.5           A Python library to read/write Excel 2010 xlsx/xlsm files
opentelemetry-api                  1.16.0          OpenTelemetry Python API
opentelemetry-sdk                  1.16.0          OpenTelemetry Python SDK
opentelemetry-semantic-conventions 0.37b0          OpenTelemetry Semantic Conventions
optbinning                         0.19.0          OptBinning: The Python Optimal Binning library
ortools                            9.10.4067       Google OR-Tools python libraries and modules
osqp                               0.6.7.post1     OSQP: The Operator Splitting QP Solver
packaging                          24.1            Core utilities for Python packages
pandas                             2.2.2           Powerful data structures for data analysis, time series, and statistics
pathspec                           0.12.1          Utility library for gitignore style pattern matching of file paths.
patsy                              0.5.6           A Python package for describing statistical models and for building desig...
pillow                             10.4.0          Python Imaging Library (Fork)
platformdirs                       4.3.3           A small Python package for determining appropriate platform-specific dirs...
pluggy                             1.5.0           plugin and hook calling mechanisms for python
pre-commit                         3.8.0           A framework for managing and maintaining multi-language pre-commit hooks.
proto-plus                         1.24.0          Beautiful, Pythonic protocol buffers.
protobuf                           5.28.1          
py4j                               0.10.9.7        Enables Python programs to dynamically access arbitrary Java objects
pyarrow                            16.1.0          Python library for Apache Arrow
pyasn1                             0.6.1           Pure-Python implementation of ASN.1 types and DER/BER/CER codecs (X.208)
pyasn1-modules                     0.4.1           A collection of ASN.1-based protocols modules
pycparser                          2.22            C parser in Python
pydantic                           2.9.1           Data validation using Python type hints
pydantic-core                      2.23.3          Core functionality for Pydantic validation and serialization
pyparsing                          3.1.4           pyparsing module - Classes and methods to define and execute parsing gram...
pytest                             8.3.3           pytest: simple powerful testing with Python
pytest-cov                         5.0.0           Pytest plugin for measuring coverage.
pytest-mock                        3.14.0          Thin-wrapper around the mock package for easier use with pytest
python-dateutil                    2.9.0.post0     Extensions to the standard Python datetime module
pytz                               2024.2          World timezone definitions, modern and historical
pyyaml                             6.0.2           YAML parser and emitter for Python
qdldl                              0.1.7.post4     QDLDL, a free LDL factorization routine.
requests                           2.32.3          Python HTTP for Humans.
ropwr                              1.0.0           RoPWR: Robust Piecewise Regression
rsa                                4.9             Pure-Python RSA implementation
s3transfer                         0.10.2          An Amazon S3 Transfer Manager
scikit-learn                       1.5.2           A set of python modules for machine learning and data mining
scipy                              1.14.1          Fundamental algorithms for scientific computing in Python
scs                                3.2.7           Splitting conic solver
seaborn                            0.13.2          Statistical data visualization
setuptools                         72.2.0          Easily download, build, install, upgrade, and uninstall Python packages
six                                1.16.0          Python 2 and 3 compatibility utilities
smmap                              5.0.1           A pure Python implementation of a sliding window memory map manager
sniffio                            1.3.1           Sniff out which async library your code is running under
sqlalchemy                         2.0.35          Database Abstraction Library
sqlparse                           0.5.1           A non-validating SQL parser.
statsmodels                        0.14.3          Statistical computations and models for Python
threadpoolctl                      3.5.0           threadpoolctl
thrift                             0.20.0          Python bindings for the Apache Thrift RPC system
typeguard                          4.3.0           Run-time type checker for Python
types-pyyaml                       6.0.12.20240917 Typing stubs for PyYAML
types-requests                     2.32.0.20240914 Typing stubs for requests
typing-extensions                  4.12.2          Backported and Experimental Type Hints for Python 3.8+
tzdata                             2024.1          Provider of IANA time zone data
urllib3                            2.2.3           HTTP library with thread-safe connection pooling, file post, and more.
virtualenv                         20.26.4         Virtual Python Environment builder
werkzeug                           3.0.4           The comprehensive WSGI web application library.
wrapt                              1.16.0          Module for decorators, wrappers and monkey patching.
zipp                               3.20.2          Backport of pathlib-compatible object wrapper for zip files
```

**Comment by fmind** [comment]
Hi @tomcorten . Thanks for the detailed explanation, I'll have a look.

**Comment by fmind** [comment]
closed by #30 

Thanks for your contribution @tomcorten , it should be released soon in v3.0.0
